Classes/Controller/EmployeeController.php

```diff
@@ -11,153 +11,63 @@
-
-    /**
-     * @var ExtConf
-     */
-    protected $extConf;
-
-    /**
-     * @var HashService
-     */
-    protected $hashService;
-
-    /**
-     * @var EmailService
-     */
-    protected $emailService;
-
-    public function injectEmployeeRepository(EmployeeRepository $employeeRepository): void
-    {
-        $this->employeeRepository = $employeeRepository;
-    }
-
-    public function injectBuildingRepository(BuildingRepository $buildingRepository): void
-    {
-        $this->buildingRepository = $buildingRepository;
-    }
-
-    public function injectDepartmentRepository(DepartmentRepository $departmentRepository): void
-    {
-        $this->departmentRepository = $departmentRepository;
-    }
-
-    public function injectOfficeRepository(OfficeRepository $officeRepository): void
-    {
-        $this->officeRepository = $officeRepository;
-    }
-
-    public function injectLanguageRepository(LanguageRepository $languageRepository): void
-    {
-        $this->languageRepository = $languageRepository;
-    }
-
-    public function injectSubjectFieldRepository(SubjectFieldRepository $subjectFieldRepository): void
-    {
-        $this->subjectFieldRepository = $subjectFieldRepository;
-    }
-
-    public function injectCategoryRepository(CategoryRepository $categoryRepository): void
-    {
-        $this->categoryRepository = $categoryRepository;
-    }
-
-    public function injectExtConf(ExtConf $extConf): void
-    {
-        $this->extConf = $extConf;
-    }
-
-    public function injectHashService(HashService $hashService): void
-    {
-        $this->hashService = $hashService;
-    }
-
-    public function injectEmailService(EmailService $emailService): void
-    {
-        $this->emailService = $emailService;
-    }
-
-    public function initializeAction(): void
-    {
-        // if this value was not set, then it will be filled with 0
-        // but that is not good, because UriBuilder accepts 0 as pid, so it's better to set it to null
-        if (empty($this->settings['pidOfDetailPage'])) {
-            $this->settings['pidOfDetailPage'] = null;
-        }
-    }
+    use InjectCategoryRepositoryTrait;
+    use InjectBuildingRepositoryTrait;
+    use InjectDepartmentRepositoryTrait;
+    use InjectEmailServiceTrait;
+    use InjectEmployeeRepositoryTrait;
+    use InjectExtConfTrait;
+    use InjectLanguageRepositoryTrait;
+    use InjectOfficeRepositoryTrait;
+    use InjectPropertyMappingConfiguratorTrait;
+    use InjectSubjectFieldRepositoryTrait;
+    use InitializeControllerActionTrait;
+    use MediaTypeConverterTrait;
+
+    public function initializeListAction(): void
+    {
+        $this->emitEventSignal();
+    }
+
+    public function listAction(array $search = []): ResponseInterface
+    {
+        $office = $this->settings['filterByOffice'] ?? null;
+        $this->postProcessAndAssignFluidVariables([
+            'records' => $this->employeeRepository->findFilteredBy($office, $search),
```

Comment:
Just `records`? Why? Please use "employees" so that every integrator knows about the datatype in fluid. Please update that variable in fluid, too.